controllers/backstage_deployment.go

```diff
@@ -312,3 +288,30 @@ func (r *BackstageReconciler) setDefaultDeploymentImage(deployment *appsv1.Deplo
 		}
 	})
 }
+
+func (r *BackstageReconciler) applyBackstageLabels(backstage bs.Backstage, deployment *appsv1.Deployment) {
+	setBackstageAppLabel(&deployment.Spec.Template.ObjectMeta.Labels, backstage)
+	setBackstageAppLabel(&deployment.Spec.Selector.MatchLabels, backstage)
+	r.labels(&deployment.ObjectMeta, backstage)
+}
+
+func (r *BackstageReconciler) applyApplicationParamsFromCR(backstage bs.Backstage, deployment *appsv1.Deployment) {
+	if backstage.Spec.Application != nil {
+		deployment.Spec.Replicas = backstage.Spec.Application.Replicas
+		if backstage.Spec.Application.Image != nil {
+			visitContainers(&deployment.Spec.Template, func(container *v1.Container) {
+				container.Image = *backstage.Spec.Application.Image
+			})
+		}
+		if backstage.Spec.Application.ImagePullSecrets != nil { // use image pull secrets from the CR spec
+			deployment.Spec.Template.Spec.ImagePullSecrets = nil
```

Comment:
This is because deployment.Spec.Template.Spec.ImagePullSecrets and backstage.Spec.Application.ImagePullSecrets have different data types.

Reply:
Since in other places for configmaps and secrets we chose to use our own types (string) instead of the standard one, I think it makes sense we use the string type here for consistency, or we may want to change all such fields to use the standard one instead.